### PR TITLE
filter: return error if passed invalid regex pattern

### PR DESCRIPTION
**Pull request by amyangfei**
### What problem does this PR solve? 

Fix issue https://github.com/pingcap/tidb-tools/issues/272

### What is changed and how it works?

Use `regexp.Compile` to replace `regexp.MustCompile`, and return the error if necessary. Note this pr will change the return list of `pkg/filter.New` 

### Check List 

Tests 

 - Unit test
 - Integration test

Code changes

 - Has exported function/method change

Side effects

Related changes

### CODE REVIEW

File: pkg/filter/filter.go [hunk 1]
```diff
@@ -114,35 +116,55 @@ func New(caseSensitive bool, rules *Rules) *Filter {
 	f.c = &cache{
 		items: make(map[string]ActionType),
 	}
-	f.genRegexMap()
-	return f
+	err := f.genRegexMap()
+	return f, err
```
Comment (csuzhangxc) [inline pkg/filter/filter.go]: if `err != nil`, the behavior of `f` is unpredictable, should we need to return `nil` rather than `f`?

**Review by WangXiangUSTC — COMMENTED** [review]
LGTM

**Review by csuzhangxc — APPROVED** [review]
LGTM

**Pull request merged**